homeassistant/components/camera/local_file.py

```diff
@@ -61,7 +75,28 @@ def camera_image(self):
             _LOGGER.warning("Could not read camera %s image from file: %s",
                             self._name, self._file_path)
 
+    def check_file_path_access(self, file_path):
+        """Check that filepath given is readable."""
+        if not os.access(file_path, os.R_OK):
+            _LOGGER.warning("Could not read camera %s image from file: %s",
+                            self._name, file_path)
+
+    def update_file_path(self, file_path):
+        """Update the file_path."""
+        self.check_file_path_access(file_path)
+        self._file_path = file_path
+        self.schedule_update_ha_state()
+
     @property
     def name(self):
         """Return the name of this camera."""
         return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the camera state attributes."""
+        attr = {
+            'access_token': self.access_tokens[-1],
```

Comment:
If the base `Camera` entity class changes anything about the state attributes or changes the access token behavior, this method would need to be updated too.

If we have inheritance here, why not use it to our advantage?

```python
attr = super(LocalFile, self).device_state_attributes
return {
  'file_path': self._file_path,
  **attr
}
```

Reply:
Great suggestion!